File: libs/ui_canvas/src/qt/preview_settings_dialog.cpp
// UI-15 — modal preview-settings container
// (ui/preview_settings_dialog.py parity).

#include <pwb/ui_canvas/qt/preview_settings_dialog.hpp>

#include <QPushButton>
#include <QVBoxLayout>

#include <pwb/ui_pages_preview/qt/preview_settings_panel.hpp>

namespace pwb::ui_canvas {

PreviewSettingsDialog::PreviewSettingsDialog(
    QWidget* parent, pwb::ui_pages_preview::PreviewSettingsStore* store)
    : QDialog(parent) {
    setObjectName(QStringLiteral("PreviewSettingsDialog"));
    setWindowTitle(QStringLiteral("预览设置"));
    setModal(true);
    setMinimumWidth(520);

    auto* layout = new QVBoxLayout(this);
    // tokens.SPACE_4 = 20 (legacy「较松区块距」).
    layout->setContentsMargins(20, 20, 20, 20);
    panel_ = new pwb::ui_pages_preview::PreviewSettingsPanel(this, store);
    layout->addWidget(panel_);

    connect(panel_,
            &pwb::ui_pages_preview::PreviewSettingsPanel::settings_applied,
            this, &PreviewSettingsDialog::settings_applied);
    // Apply accepts; Reset only mutates settings (panel-internal).
    connect(panel_->apply_button(), &QPushButton::clicked, this,
            &PreviewSettingsDialog::accept);
}

void PreviewSettingsDialog::set_settings(
    const pwb::ui_pages_preview::PreviewSettings& settings) {
    panel_->set_settings(settings);
}

void PreviewSettingsDialog::set_preview_mode(const QString& mode) {
    panel_->set_preview_mode(mode);
}

}  // namespace pwb::ui_canvas

File: legacy/python_reference/product/paleo_workbench/ui/preview_settings_dialog.py
from __future__ import annotations

from PySide6.QtCore import Signal
from PySide6.QtWidgets import QDialog, QVBoxLayout

from paleo_workbench.ui import tokens
from paleo_workbench.ui.pages.preview_settings import (
    PreviewSettings,
    PreviewSettingsStore,
)
from paleo_workbench.ui.pages.preview_settings_panel import PreviewSettingsPanel


class PreviewSettingsDialog(QDialog):
    """Application-level modal container for the shared preview editor."""

    settings_applied = Signal(object)

    def __init__(
        self,
        parent=None,
        *,
        store: PreviewSettingsStore | None = None,
    ) -> None:
        super().__init__(parent)
        self.setObjectName("PreviewSettingsDialog")
        self.setWindowTitle("预览设置")
        self.setModal(True)
        self.setMinimumWidth(520)

        layout = QVBoxLayout(self)
        layout.setContentsMargins(
            tokens.SPACE_4,
            tokens.SPACE_4,
            tokens.SPACE_4,
            tokens.SPACE_4,
        )
        self.panel = PreviewSettingsPanel(store=store)
        layout.addWidget(self.panel)

        self.panel.settings_applied.connect(self.settings_applied)
        self.panel.apply_btn.clicked.connect(self.accept)

    def set_settings(self, settings: PreviewSettings) -> None:
        self.panel.set_settings(settings)

    def set_preview_mode(self, mode: str) -> None:
        self.panel.set_preview_mode(mode)
